### PR TITLE
feat: Bump minimum Java version to 11

### PR DESCRIPTION
**Pull request by Artur-**

### CODE REVIEW

**Comment by manolo** [comment]
Agents in TC report : `Java version: 1.8.0_201, vendor: Oracle Corporation, runtime: /usr/lib/jvm/jdk1.8.0_201/jre`. 
What should we do here? 
- upgrade agents 
- modify builds so as they can detect flow version and set the appropriate javaHome

**Comment by Artur-** [comment]
Stop using the TC build for PRs?

**Comment by manolo** [comment]
Modified TC validations so as they run java11 in the case branch has these changes.

File: pom.xml [hunk 1]
```diff
@@ -68,13 +68,13 @@
             https://maven.vaadin.com/vaadin-addons
         </flow.addons.repo.url>
 
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
```
Comment (manolo) [inline pom.xml:72]: What about producing bytecode for java 8 ? but still have java 11 as the source ?

Comment (Artur-) [reply pom.xml:72]: Why would we do that?

Comment (manolo) [reply pom.xml:72]: So as they can still deploy vaadin in their old systems.

Comment (Artur-) [reply pom.xml:72]: That would never work as you cannot use e.g. `String.strip` and any other Java API added after 8. It would be constantly broken.

**Review by manolo — APPROVED** [review]

**Pull request merged**